cns/service/main.go

```diff
@@ -1303,6 +1303,52 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 	// TODO(rbtr): nodename and namespace should be in the cns config
 	directscopedcli := nncctrl.NewScopedClient(directnnccli, types.NamespacedName{Namespace: "kube-system", Name: nodeName})
 
+	// Create the base NNC CRD if HomeAz is enabled
+	if cnsconfig.EnableHomeAz {
+		homeAzResponse := httpRestServiceImplementation.GetHomeAz(ctx)
+		az := homeAzResponse.HomeAzResponse.HomeAz
+		logger.Printf("[Azure CNS] HomeAz: %d", az)
+		// Create Node Network Config CRD and update the Home Az field with the cache value from the HomeAz Monitor
+		var nnc *v1alpha.NodeNetworkConfig
+		if nnc, err = directnnccli.Get(ctx, types.NamespacedName{Namespace: "kube-system", Name: nodeName}); err != nil {
+			logger.Errorf("[Azure CNS] failed to get existing NNC: %v", err)
+		}
+
+		newNNC := createBaseNNC(node)
+		if nnc == nil {
+			logger.Printf("[Azure CNS] Creating new base NNC")
+			newNNC.Spec.AvailabilityZone = az
+			if err = directcli.Create(ctx, newNNC); err != nil {
+				return errors.Wrap(err, "failed to create base NNC")
+			}
+		} else {
+			logger.Printf("[Azure CNS] Patching existing NNC with new Spec with HomeAz %d", az)
+			newNNC.Spec.AvailabilityZone = az
+			newNNC.Spec.RequestedIPCount = nnc.Spec.RequestedIPCount
+			newNNC.Spec.IPsNotInUse = nnc.Spec.IPsNotInUse
+			newNNC.Status = nnc.Status
+			newNNC.UID = nnc.UID
+			newNNC.Name = nnc.Name
+			newNNC.Namespace = nnc.Namespace
+			newNNC.Annotations = nnc.Annotations
+			newNNC.Labels = nnc.Labels
+			newNNC.Finalizers = nnc.Finalizers
+			newNNC.OwnerReferences = nnc.OwnerReferences
+			newNNC.CreationTimestamp = nnc.CreationTimestamp
+			newNNC.DeletionTimestamp = nnc.DeletionTimestamp
+
+			// Delete existing NNC and create new one with updated HomeAz
+			if err = directcli.Delete(ctx, nnc); err != nil {
+				return errors.Wrap(err, "[Azure CNS]: failed to delete existing NNC")
+			}
+
+			if err = directcli.Create(ctx, newNNC); err != nil {
+				return errors.Wrap(err, "[Azure CNS]: failed to create new NNC")
+			}
```

Comment:
Is there no update method? Deleting and recreating seems very prone to issues.

Reply:
I was running into issue trying to simply Patch the NNC. So after testing went this route. Let me retry again. Tried a few combinations of testing. Also attempted to simply update the az field on the existing NNC but that is possible only once the new NNC changes in DNC-RC are rolled out first before these updates.

Reply:
I was able to use the Update method effectively by using the resourceID from the existing NNC so updated to use that API.